tensorboard/java/org/tensorflow/tensorboard/vulcanize/Vulcanize.java

```diff
@@ -778,6 +778,79 @@ private static String extractAndTransformJavaScript(Document document, Webpath j
     return scriptContent;
   }
 
+  private static void cloneChildrenWithoutWhitespace(Element src, Element dest) {
+    List<Node> toMove = new ArrayList<Node>();
+    for (Node node : src.childNodes()) {
+      if (node instanceof TextNode && ((TextNode) node).isBlank()) {
+        continue;
+      }
+      toMove.add(node);
+    }
+    for (int i = 0; i < toMove.size(); i++) {
+      Node node = toMove.get(i);
+      dest.appendChild(node.clone());
+    }
+  }
+
+  // When we inline the HTML based on `<link rel="import">` in `transform`, we
+  // replace the link element with parsed document. This makes us have nested
+  // documents and jsoup's Node.outerHtml (or Node.toString) are incapable of
+  // properly outputting that. Here, we flatten the document by combining all
+  // elements in `<head>` and `<body>` of nested document in one `<head>` and
+  // `<body>`.
+  //
+  // Examples:
+  // // Input
+  // <#root> <!-- document -->
+  //   <html>
+  //     <head><#root><html><body>foo</body></html></#root></head>
```

Comment:
I think it would be worth expanding the example (or add another) to include a nested `head` and nested `body` element, if those are actual things we have to worry about (I think they are).